app/services/listing_service/store/shape.rb

```diff
@@ -116,7 +117,9 @@ def to_tt_model_attributes(hash)
       raise ArgumentError.new("Unknown unit type #{unit_type}")
     end
 
-    model_hash.except(:units)
+    model_hash[:url] = uniq_url(hash[:url_source], hash[:community_id])
```

Comment:
I would move this up into `create` and `update` methods.

The purpose of `to_tt_model_attributes` and `from_tt_model_attributes` was to map outdated model field names to updated API field names, so that:

`from_tt_model_attributes(to_tt_model_attributes(hash)) == hash`

 My idea is to remove both of these methods in the future. When we rename the `TransactionType` to `ListingShape` we also rename the `price_field` to `price_enabled` and in the end we remove the need for this method altogether.